client_test.go

```diff
@@ -0,0 +1,30 @@
+package client_test
+
+import (
+	"testing"
+
+	mfClient "github.com/manifestival/client-go-client"
+	mf "github.com/manifestival/manifestival"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	kFake "k8s.io/client-go/dynamic/fake"
+)
+
+func TestDynamicClient(t *testing.T) {
+	scheme := runtime.NewScheme()
+
+	kfdc := kFake.NewSimpleDynamicClient(scheme)
+	client, err := mfClient.NewDynamicClient(kfdc)
+
+	if err != nil {
+		t.Fatalf("received client creation error %v", err)
+	}
+
+	source := mf.Slice{}
+	_, err = mf.ManifestFrom(source, mf.UseClient(client))
+
+	if err != nil {
+		t.Fatalf("received creating manifest %v", err)
+	}
+	// TODO: validate calls with objects in fake dynamic client
```

Comment:
I can do this later for both impls if you're busy now.

Reply:
🥇 Great! I'd definitely take you up on that.

Testing with the unsafe dynamic client will probably be easier as you should be able to pass in the objects directly using "k8s.io/client-go/dynamic/fake": https://github.com/knative/serving/blob/d947bf413d5395f2182a2127beb68b23fd9ed2d0/pkg/reconciler/testing/v1/factory.go#L73
https://github.com/knative/pkg/blob/ffb929374a39cc0659b18100e9e2361f5106a81f/injection/clients/dynamicclient/fake/fake.go#L41